### PR TITLE
Wait stacks to be ready before changing traffic

### LINKED ISSUE #398: Senza traffic should wait for stack to be ready to change traffic
### Current Behaviour

When running change traffic `senza traffic` on stack recently deployed. The status code of the stack is not ready (`(CRETE|UPDATE)_COMPLETE`) and Senza fails trying to update it.
### Cause of the Problem

It is not possible to update CF stack when it's status code is not in `(CRETE|UPDATE)_COMPLETE` (or we should not change when it is not in one of those status code).
### Expected Behaviour

Senza should wait the stack to get in a final status code if the current status of the stack is `(CREATE|UPDATE)_IN_PROGRESS` and ending up in some good state `(CREATE|UPDATE)_COMPLETE`.

### PR DESCRIPTION
**Pull request by rafaelcaricio**
Fixes #398

### CODE REVIEW

File: senza/cli.py [hunk 1]
```diff
@@ -1152,29 +1153,32 @@ def traffic(stack_name, stack_version, percentage, region, output):
...
+                    if len(related_stacks) > 0:
+                        for related_stack in related_stacks:
+                            current_stack_status = related_stack.StackStatus
+
+                            if current_stack_status.endswith('_COMPLETE') or current_stack_status.endswith('_FAILED'):
+                                continue
+                            elif current_stack_status.endswith('_IN_PROGRESS'):
+                                # some operation in progress, let's wait some time to try again
+                                all_stacks_in_final_state = False
+                                info(
```
Comment (lukasniemeier-zalando) [inline senza/cli.py]: This can get a bit spammy right? :wink:

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by rafaelcaricio** [comment]
👍

File: senza/cli.py (hunk 1 above)
Comment (rafaelcaricio) [inline senza/cli.py]: @lukasniemeier-zalando now I have added an option to be able to set the wait interval. So you control the spammy behaviour. ;)

File: senza/cli.py [hunk 2]
```diff
@@ -1152,29 +1156,32 @@ def traffic(stack_name, stack_version, percentage, region, output):
...
+                                    "Waiting stack {} ({}) to perform traffic change...".format(
```
Comment (hjacobs) [inline senza/cli.py]: The grammar sounds wrong: better: "Waiting for stack ... to perform traffic change.." (add "for" and use only two trailing dots as we do everywhere)

**Comment by hjacobs** [comment]
@rafaelcaricio you need to fix tests too..

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**